portfolio/src/main/webapp/script.js

```diff
@@ -11,7 +11,65 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
- 
+
+
+// These variables will be used to track whenever a reply button is clicked
+// And whether the same button was clicked again in a row.
+var replyClickCounter = 0;
+var currentReplyBtnClicked = 0;
+
+// Post event listener to submit posts with Fetch on form submit. 
+document.getElementById('post-form').addEventListener('submit', function(postSubmitEvent) {
+  // Prevent form from redirecting to the server.
+  postSubmitEvent.preventDefault();
+  
+  // Get a reference to all the elements on the form.
+  const formData = new FormData(this);
+  const postFormParams = new URLSearchParams();
+  
+  // Append a pair [parameter name: value] of the form's elements
+  // to a URLSearchParams.
+  for (const pair of formData) {
+    postFormParams.append(pair[0], pair[1]);
+  }
+
+  // Make a POST request with the new post and fetch it to add to the DOM.
+  fetch('/data', {method: 'POST', body: postFormParams}).then(
+        response => response.json()).then((msgs) => {
+    const statsListElement = document.getElementById('posts-list');
+    msgs.forEach((msg) => {
+      statsListElement.appendChild(createListElement(msg));
+    })
+  }).catch (function(error) {
+      console.log(error);
+      // No form reset if post wasn't submitted; return.
+      return;
+  })
+  
+  // Empty the post form after a post is submitted.
+  resetForm('post-form');
+});
+
+// Reply event listener to submit replies with Fetch on form submit.
+document.getElementById('reply-form').addEventListener('submit', function(replySubmitEvent) {
+  replySubmitEvent.preventDefault();
+
+  const formData = new FormData(this);
+  const replyFormParams = new URLSearchParams();
+  
+  for (const pair of formData) {
+    replyFormParams.append(pair[0], pair[1]);
+  }
```

Comment:
Same point here as above... it would make more sense to append the URLSearchParams to the formData (if possible).